File: RealEstateAnalysis.Domain/DTOs/Writes/RentalProperty/WriteProperty.cs
using Microsoft.AspNetCore.Http;
using System.Collections.Generic;

namespace RealEstateAnalysis.Domain.DTOs.Writes.RentalProperty
{
    public class WriteProperty
    {
        public WriteAddress Address { get; set; }

        public decimal AnnualGrossScheduledRentalIncome { get; set; }

        public decimal AnnualGrossScheduledRentalIncomeGrowthRate { get; set; }

        public List<WriteAnnualOperatingExpense> AnnualOperatingExpenses { get; set; }

        public decimal AnnualOperatingExpensesGrowthRate { get; set; }

        public decimal AnnualPropertyManagementFeeRate { get; set; }

        public decimal AnnualVacancyRate { get; set; }

        public int BuildingSquareFootage { get; set; }

        public List<WriteClosingCost> ClosingCosts { get; set; }

        public decimal DownPayment { get; set; }

        public List<WriteExteriorRepairExpense> ExteriorRepairExpenses { get; set; }

        public List<WriteGeneralRepairExpense> GeneralRepairExpenses { get; set; }

        public string GroupName { get; set; }

        public List<WriteInteriorRepairExpense> InteriorRepairExpenses { get; set; }

        public decimal LoanApr { get; set; }

        public int LoanYears { get; set; }

        public int LotSquareFootage { get; set; }

        public decimal MarketCapitalizationRate { get; set; }

        public string Notes { get; set; }

        public decimal OtherAnnualIncome { get; set; }

        public long PropertyTypeId { get; set; }

        public long PropertyStatusId { get; set; }

        public decimal PurchasePrice { get; set; }

        public IFormFile ThumbnailImage { get; set; }

        public List<WriteUnitGroup> UnitGroups { get; set; }

        public int YearBuiltIn { get; set; }
    }
}

File: RealEstateAnalysis.Web/ClientApp/src/app/modules/rental-properties/dtos/writes/writeProperty.ts
import { WriteAnnualOperatingExpense } from './writeAnnualOperatingExpense';
import { WriteExteriorRepairExpense } from '../../../../shared/dtos/writes/writeExteriorRepairExpense';
import { WriteGeneralRepairExpense } from '../../../../shared/dtos/writes/writeGeneralRepairExpense';
import { WriteAddress } from '../../../../shared/dtos/writes/writeAddress';
import { WriteInteriorRepairExpense } from '../../../../shared/dtos/writes/writeInteriorRepairExpense';
import { WriteClosingCost } from '../../../../shared/dtos/writes/writeClosingCost';
import { WriteUnitGroup } from '../../../../shared/dtos/writes/writeUnitGroup';

export class WriteProperty {
    constructor(public thumbnailImage: File | null,
        public yearBuiltIn: number,
        public buildingSquareFootage: number,
        public lotSquareFootage: number,
        public propertyTypeId: number,
        public propertyStatusId: number,
        public notes: string,
        public address: WriteAddress,
        public unitGroups: WriteUnitGroup[],
        public annualOperatingExpenses: WriteAnnualOperatingExpense[],
        public annualOperatingExpensesGrowthRate: number,
        public marketCapitalizationRate: number,
        public annualGrossScheduledRentalIncomeGrowthRate: number,
        public annualGrossScheduledRentalIncome: number,
        public annualVacancyRate: number,
        public annualPropertyManagementFeeRate: number,
        public closingCosts: WriteClosingCost[],
        public downPayment: number,
        public exteriorRepairExpenses: WriteExteriorRepairExpense[],
        public generalRepairExpenses: WriteGeneralRepairExpense[],
        public interiorRepairExpenses: WriteInteriorRepairExpense[],
        public loanApr: number,
        public loanYears: number,
        public otherAnnualIncome: number,
        public purchasePrice: number,
        public groupName: string) { }
}
